src/app/page.tsx

```diff
@@ -1,116 +1,78 @@
-	const contract = getContract({
-		address: defaultNftContractAddress,
-		chain,
-		client,
-	});
-	const isERC721Query = useReadContract(isERC721, { contract });
-	const isERC1155Query = useReadContract(isERC1155, { contract });
-	const contractMetadataQuery = useReadContract(getContractMetadata, {
-		contract,
-	});
-
-	const nftQuery = useReadContract(getNFT, {
-		contract,
-		tokenId,
-		queryOptions: { enabled: isERC1155Query.data },
-	});
-
-	const claimCondition1155 = useReadContract(getActiveClaimCondition1155, {
-		contract,
-		tokenId,
-		queryOptions: {
-			enabled: isERC1155Query.data,
-		},
-	});
-
-	const claimCondition721 = useReadContract(getActiveClaimCondition721, {
-		contract,
-		queryOptions: { enabled: isERC721Query.data },
-	});
+// lib imports for fetching NFT details
+import { getERC20Info } from "@/lib/erc20";
+import { getERC721Info } from "@/lib/erc721";
+import { getERC1155Info } from "@/lib/erc1155";
+// thirdweb imports
+import { defineChain, getContract } from "thirdweb";
+import { isERC1155 } from "thirdweb/extensions/erc1155";
+import { isERC721 } from "thirdweb/extensions/erc721";
 
-	const claimCondition20 = useReadContract(getActiveClaimCondition20, {
-		contract,
-		queryOptions: { enabled: !isERC721Query.data && !isERC1155Query.data },
-	});
+const chain = defineChain(defaultChainId);
+const contract = getContract({ address: defaultNftContractAddress, chain, client });
 
-	const displayName = isERC1155Query.data
-		? nftQuery.data?.metadata.name
-		: contractMetadataQuery.data?.name;
+async function getERCType() {
+  try {
+    const [isErc721, isErc1155] = await Promise.all([
+      isERC721({ contract }).catch(() => false),
+      isERC1155({ contract }).catch(() => false),
+    ]);
 
-	const description = isERC1155Query.data
-		? nftQuery.data?.metadata.description
-		: contractMetadataQuery.data?.description;
-
-	const priceInWei = isERC1155Query.data
-		? claimCondition1155.data?.pricePerToken
-		: isERC721Query.data
-			? claimCondition721.data?.pricePerToken
-			: claimCondition20.data?.pricePerToken;
-
-	const currency = isERC1155Query.data
-		? claimCondition1155.data?.currency
-		: isERC721Query.data
-			? claimCondition721.data?.currency
-			: claimCondition20.data?.currency;
-
-	const currencyContract = getContract({
-		address: currency || "",
-		chain,
-		client,
-	});
+    return isErc1155 ? "ERC1155" : isErc721 ? "ERC721" : "ERC20";
+  } catch (error) {
+    console.error("Error detecting ERC type:", error);
+    return null;
+  }
+}
 
-	const currencyMetadata = useReadContract(getCurrencyMetadata, {
-		contract: currencyContract,
-		queryOptions: { enabled: !!currency },
-	});
+export default async function Home() {
+  try {
+    const ercType = await getERCType();
+    if (!ercType) throw new Error("Failed to determine ERC type.");
 
-	const currencySymbol = currencyMetadata.data?.symbol || "";
+    // Ensure errors are caught properly by using await inside try/catch
```

Comment:
this comment is not that useful, should say something about thirdweb like

`// fetch contract information depending on the ERC type`